libs/widget-lib/src/IframeRpcProviderBridge.ts

```diff
@@ -74,53 +81,53 @@ export class IframeRpcProviderBridge {
    * Processes a JSON-RPC request and sends appropriate response or error via the content window.
    * @param request - The JSON-RPC request to be processed.
    */
-  processRequest(request: JsonRpcRequest) {
-    if (!this.ethereumProvider) return
-
+  processRpcRequest(request: JsonRpcRequestMessage) {
+    const { id, jsonrpc, method } = request
+    if (!this.ethereumProvider || !id) {
+      return
+    }
     const requestPromise =
-      request.method === 'enable' ? this.ethereumProvider.enable() : this.ethereumProvider.request(request)
+      method === 'enable' ? this.ethereumProvider.enable() : this.ethereumProvider.request({ ...request, id })
 
     // Do request, and forward the result or error to the iFrame window
     requestPromise
-      .then((result) => this.forwardRpcCallToIframe({ id: request.id, result }))
-      .catch((error) => this.forwardRpcCallToIframe({ id: request.id, error }))
+      .then((result) =>
+        this.forwardRpcResponseToIframe({
+          rpcResponse: { jsonrpc, id, result },
+        })
+      )
+      .catch((error) =>
+        this.forwardRpcResponseToIframe({
+          rpcResponse: { jsonrpc, id, error },
+        })
+      )
   }
 
-  private processRpcCallFromWindow = (event: MessageEvent): void => {
-    if (!isRpcCall(event)) {
-      return
-    }
-
+  private processRpcCallFromWindow = ({ rpcRequest }: ProviderRpcRequestPayload): void => {
     // If disconnected, ignore the message. Also persist the request
     if (!this.ethereumProvider) {
-      if (event.data.id) {
-        this.requestWaitingForConnection[event.data.id] = event.data
+      if (rpcRequest.id) {
+        this.requestWaitingForConnection[rpcRequest.id] = rpcRequest
       }
       return
     }
 
-    // Delegate the rpc call
-    this.processRequest(event.data)
+    this.processRpcRequest(rpcRequest)
   }
 
-  private onRpcEventForwardToIframe(event: string, params: unknown): void {
-    this.forwardRpcCallToIframe({ method: event, params: [params] })
+  private onProviderEvent(event: string, params: unknown): void {
+    // console.debug('[IframeRpcProviderBridge] onProviderEvent', { event, params })
```

Comment:
More debug logs. I'll stop mentioning them

Reply:
Deleted this and all of the other dead codes for console messages (from this PR and others) https://github.com/cowprotocol/cowswap/pull/3911